contrib/completion/bash/docker

```diff
@@ -201,19 +201,36 @@ __docker_signals() {
 
 _docker_docker() {
 	local boolean_options="
-		--daemon -d
 		--debug -D
 		--help -h
+		--tls
+		--tlsverify
+		--tlscacert
+		--tlscert
+		--tlskey
+		--version -v
+	"
+
+	case "$cur" in
+		-*)
+			COMPREPLY=( $( compgen -W "$boolean_options $main_options_with_args" -- "$cur" ) )
```

Comment:
`$main_options_with_args` still contains the previous set of top-level arguments, including those needed for daemon mode. Therefore, completion will wrongly support things like `docker --api-cors-header`.
Please reduce the definition of `$main_options_with_args` below to the actual global options that take args: 

```
--host -H --log-level -l --tlscacert --tlscert --tlskey
```